PyStemmusScope/save.py

```diff
@@ -0,0 +1,334 @@
+    data = data.astype('float32')
+
+    # convert dataframe to xarray data array
+    return data.to_xarray()
+
+
+def _create_soil_layer_metadata(thicknesses: List, depths: List) -> List:
+    """
+    layer_1: 0.0 - 1.0 cm
+    layer_2: 1.0 - 2.0 cm
+    layer_3: 2.0 - 3.0 cm
+    """
+
+    metadata = []
+    for index, (thickness, depth) in enumerate(zip(thicknesses, depths)):
+        metadata.append(f"layer_{index + 1}: {(depth - thickness)} - {depth} cm")
+
+    return metadata
+
+
+def _update_dataset_attrs_dims(dataset: xr.Dataset, forcing_dict: Dict) -> xr.Dataset:
+    """Update dimentions of a dataset according to ALMA conventions.
+
+    Args:
+        dataset(xr.Dataset): a dataset with varaibles in ALMA conventions.
+
+    Returns:
+        xr.Dataset: the dataset with dimensions ("time", "x", "y").
+    """
+
+    # add x/y dims to the dataset
+    dataset_expanded = dataset.expand_dims(["x", "y"])
+
+    # change the order of dims
+    try:
+        dataset_reordered = dataset_expanded.transpose("time", "y", "x", "z")
+    except ValueError:
+        try:
+            dataset_reordered = dataset_expanded.transpose("time", "y", "x")
+        except ValueError as err:
+            raise ValueError("Data should have time dimension.") from err
+
+    # additional metadata
+    lat = forcing_dict["latitude"]
+    lon = forcing_dict["longitude"]
+    dataset_reordered.attrs = {
+        'model': 'STEMMUS_SCOPE',
+        'institution': 'University of Twente; Northwest A&F University',
+        'contact': (
+            'Zhongbo Su, z.su@utwente.nl; '
+            'Yijian Zeng, y.zeng@utwente.nl; '
+            'Yunfei Wang, y.wang-3@utwente.nl'
+            ),
+        'license_type': 'CC BY 4.0',
+        'license_url': 'https://creativecommons.org/licenses/by/4.0/',
+        'latitude': lat,
+        'longitude': lon,
+        }
+
+    # update values of x and y coords
+    dataset = dataset_reordered.assign_coords(
+        {
+            "x": [lon],
+            "y": [lat],
+            }
+        )
+
+    # update x, y attributes
+    dataset["x"].attrs = {
+            "long_name": "Gridbox longitude",
+            "standard_name": "longitude",
+            "units": "degrees",
+            }
+
+    dataset["y"].attrs = {
+            "long_name": "Gridbox latitude",
+            "standard_name": "latitude",
+            "units": "degrees",
+            }
+
+    return dataset
+
+
+def to_netcdf(config: Dict, cf_filename: str) -> str:
+    """Save csv files generated by Stemmus_Scope model to a netcdf file using
+        information provided by ALMA conventions.
+
+    Args:
+        config(Dict): PyStemmusScope configuration dictionary.
+        cf_filename(str): Path to a csv file for ALMA conventions.
+
+    Returns:
+        str: path to a csv file under the output directory.
+    """
+
+    # list of required forcing variables, Alma_short_name: forcing_io_name, # model_name
+    # they called ECdata
```

Comment:
What does this line mean?

Reply:
This line is now removed. It was a reference to the function  `generate_ec_data` [here](https://github.com/EcoExtreML/STEMMUS_SCOPE/blob/16f5c7e9ba62c864684b12f6e4b246a93520ea48/utils/csv_to_nc/csv_to_nc.py#L6).